modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -218,7 +220,8 @@ abstract class HealthCheckConsensus[
         .map(_.isDefined)
         .ifM(
           logger.info(s"Positive outcome for peer: ${key.id}. Peer is already known."),
-          logger.info(s"Positive outcome for peer: ${key.id}. Peer is already unknown.")
+          logger.info(s"Positive outcome for peer: ${key.id}. Peer is already unknown. Adding to joining queue.") >>
```

Comment:
Initially we wanted to either remove some peers or keep them in the cluster, as an outcome from the healthcheck.
If the node is unhealthy - it should be removed, kept otherwise. So for the non-negative outcome from the healthcheck, we should either keep the peer or try to rejoin it.

Reply:
I don't think that the result of the peer declaration health check should be adding nodes to the cluster, it doesn't make sense. This health check is for unlocking the consensus and kicking-out peers who are not publishing their declarations in a timely manner.

I think that if anything, joining the node on a positive outcome should be an action taken only after the ping health check and could be implemented as a hook `onPositiveOutcome` that's implemented by a ping health check only. It would also make more sense to add the `ipAddress` and `port` only to the ping health check key as these values could be actually used for pinging (or for comparing with what we have in cluster storage).

Reply:
@tbekas I just wanted to mimic the behavior that we have in mainnet 1.0 and it seems logical to me, to keep the cluster aligned. Missing declaration is just a trigger with its logic to determine the outcome. It's not an entirely different healthcheck, hence, when the peer is missing but healthcheck outcome is non-negative, it means that this peer should be in the cluster, and it doesn't matter what triggered that healthcheck. If one node does not have that peer, it should be rejoined.

> This health check is for unlocking the consensus and kicking-out peers who are not publishing their declarations in a timely manner.
Every healthcheck in our codebase is to keep the cluster aligned. Whether by removing peers that should be removed or by adding peers that should be added, thus the decision is binary - peer must be in the cluster or not. As an effect of cluster alignment, consensus will be unlocked.

@TheMMaciek what's your opinion on that?